File: app/src/main/java/com/vincent/githubusersearching/utilities/Utility.java
package com.vincent.githubusersearching.utilities;

import android.widget.Toast;

import androidx.annotation.StringRes;

import com.vincent.githubusersearching.AppController;

public class Utility {

    public static void toastShort(String msg) {
        Toast.makeText(AppController.getInstance().getApplicationContext(), msg, Toast.LENGTH_SHORT).show();
    }

    public static void toastShort(@StringRes int msgResId) {
        toastShort(AppController.getInstance().getApplicationContext().getString(msgResId));
    }

    public static void toastLong(String msg) {
        Toast.makeText(AppController.getInstance().getApplicationContext(), msg, Toast.LENGTH_LONG).show();
    }

    public static void toastLong(@StringRes int msgResId) {
        toastLong(AppController.getInstance().getApplicationContext().getString(msgResId));
    }

    public static int getTotalPageCount(int totalCount, int countPerPage) {
        int totalPage = (int) Math.ceil((double) totalCount / countPerPage);
        return totalCount > 1000 ? 1000 : totalPage;
    }
}

File: app/src/main/java/com/vincent/githubusersearch/utilities/Utility.kt
package com.vincent.githubusersearch.utilities

import android.widget.Toast

import androidx.annotation.StringRes

import com.vincent.githubusersearch.AppController
import kotlin.math.ceil

object Utility {

    fun toastShort(msg: String) {
        Toast.makeText(AppController.instance.applicationContext, msg, Toast.LENGTH_SHORT).show()
    }

    fun toastShort(@StringRes msgResId: Int) {
        toastShort(AppController.instance.applicationContext.getString(msgResId))
    }

    fun toastLong(msg: String) {
        Toast.makeText(AppController.instance.applicationContext, msg, Toast.LENGTH_LONG).show()
    }

    fun toastLong(@StringRes msgResId: Int) {
        toastLong(AppController.instance.applicationContext.getString(msgResId))
    }

    fun getTotalPageCount(totalCount: Int, countPerPage: Int): Int {
        val totalPage = ceil(totalCount.toDouble() / countPerPage).toInt()
        return if (totalCount > 1000) 1000 else totalPage
    }
}
